moondance-labs/tanssi
# Enable PoV reclaim in all runtimes

Allows to reclaim unused proof size. So if the benchmark says that an extrinsic will consume 1000 bytes because that's the worst case, but in reality it only consumes 10 bytes, the 990 bytes of difference will not count for the weight calculation and will be available for other extrinsics.

I followed this guide: https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/guides/enable_pov_reclaim/index.html

To test locally:

```
export RUST_LOG=runtime::storage_reclaim=trace
pnpm moonwall run zombie_tanssi_one_node
```

And in polkadot js, send any extrinsic, for example dataPreservers.createProfile. You will see this in node logs:

```
2024-06-12 13:08:54.022 TRACE tokio-runtime-worker runtime::storage_reclaim: [Orchestrator] Reclaiming storage weight. benchmarked: 3694, consumed: 38 unspent: 0    
```

Also updated polkadot-sdk to fix problem with litep2p dependency not compiling.
